src/module/profile/interface/profile.controller.ts

```diff
@@ -0,0 +1,32 @@
+import { Controller, Get, Param, ParseIntPipe, Body, Put, Post, UseGuards } from '@nestjs/common';
+import { ProfileService } from '../application/service/profile.service';
+import { Profile } from '../domain/profile.entity';
+import { UpdateProfileDto } from '../application/dto/update-profile.dto';
+import { CreateProfileDto } from '../application/dto/create-profile.dto';
+import { JwtGuard } from 'src/common/guards/jwt.guard';
+
+@Controller('profile')
```

Comment:
All these endpoints should allow admin role.
Only PUT and GET should be allowed for the authenticated user.